extensions/datasource/deployment/src/main/java/io/quarkus/datasource/deployment/devservices/DevServicesDatasourceProcessor.java

```diff
@@ -42,40 +39,16 @@
 import io.quarkus.deployment.pkg.builditem.CurateOutcomeBuildItem;
 import io.quarkus.runtime.LaunchMode;
 import io.quarkus.runtime.configuration.ConfigUtils;
-import io.smallrye.config.Expressions;
-import io.smallrye.config.SmallRyeConfig;
 
 @BuildSteps(onlyIfNot = IsNormal.class, onlyIf = GlobalDevServicesConfig.Enabled.class)
 public class DevServicesDatasourceProcessor {
 
     private static final Logger log = Logger.getLogger(DevServicesDatasourceProcessor.class);
     private static final int DOCKER_PS_ID_LENGTH = 12;
 
-    // list of devservices properties we should not check for restart
-    // see issue #30390
-    private static final Set<String> EXCLUDED_RESTART_PROPERTIES = Set.of();
-
-    /**
-     * Properties that if modified will trigger a datasource dev service restart
-     */
-    private static final Set<String> RESTART_TRIGGERING_PROPERTIES = Set.of("db-kind",
-            "db-version",
-            "devservices.command",
-            "devservices.container-env.",
-            "devservices.container-properties.",
-            "devservices.db-name",
-            "devservices.image-name",
-            "devservices.init-script-path",
-            "devservices.port",
-            "devservices.properties.",
-            "devservices.reuse",
-            "devservices.volumes.",
-            "password",
-            "username");
```

Comment:
I think you'll need to get `quarkus.datasource[."name"].user` and `quarkus.datasource[."name"].password` as well, given a change to these properties used to trigger a restart... and for good reason: they are used as a default for devservice user/password:

https://github.com/quarkusio/quarkus/blob/e39d84e14516ebe006e0c31531511b8b496823ad/extensions/datasource/deployment/src/main/java/io/quarkus/datasource/deployment/devservices/DevServicesDatasourceProcessor.java#L289-L292

To avoid problems with quoted/unquoted, maybe you could get the values using the exact same code used above (`getFirstOptionalValue`)?